release/apparmor.go

```diff
@@ -83,43 +94,57 @@ var (
 	// appArmorParserError contains an error, if any, encountered when
 	// discovering available parser features.
 	appArmorParserError error
+
+	appArmorOnces = new(appArmorOnceBattery)
+
+	appArmorProbeKernel = probeAppArmorKernelFeaturesOnce
+	appArmorProbeParser = probeAppArmorParserFeaturesOnce
+	appArmorAssess      = assessAppArmorOnce
 )
 
+func assessAppArmorOnce() {
+	appArmorOnces.assess.Do(assessAppArmor)
+}
+
 // AppArmorLevel quantifies how well apparmor is supported on the current
 // kernel. The computation is costly to perform. The result is cached internally.
 func AppArmorLevel() AppArmorLevelType {
-	if appArmorLevel == UnknownAppArmor {
-		assessAppArmor()
-	}
+	appArmorAssess()
```

Comment:
I looked into this, in the end I reorganized things quite a bit, mostly because the Mock* functions below are complicated/long  but didn't help fully with testing this. I'm pushing the new code now, with it commenting the Once.Do bits always make some test fail.